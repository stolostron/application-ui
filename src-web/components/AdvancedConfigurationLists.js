/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import ResourceList from './common/ResourceList'
import ResourceTableModule from './common/ResourceTableModuleFromProps'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import getResourceDefinitions from '../definitions'
import { makeGetVisibleTableItemsSelector } from '../reducers/common'
import QuerySwitcher from './common/QuerySwitcher'

const AdvancedConfigurationLists = (props) => {

  const resourceType = RESOURCE_TYPES.QUERY_APPLICATIONS
  const staticResourceData = getResourceDefinitions(resourceType)
  const getVisibleResources = makeGetVisibleTableItemsSelector(resourceType)

  return (
    <ResourceList
      {...props}
      resourceType={resourceType}
      staticResourceData={staticResourceData}
      getVisibleResources={getVisibleResources}
      modules={[<ResourceTableModule key="deployments" definitionsKey="deploymentKeys" />]}
    >
      <QuerySwitcher
        key='switcher'
        queryParam="resource"
        options={
              [{ id: 'subscription', contents: 'Subscriptions' },
                { id: 'placementrule', contents: 'Placement rules'},
                {id: 'channel', contents: 'Channels'}]}
        defaultOption="subscription"
      />
    </ResourceList>
  )}

export default AdvancedConfigurationLists
