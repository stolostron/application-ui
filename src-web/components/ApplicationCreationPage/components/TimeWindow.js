/*******************************************************************************
 * Licensed Materials - Property of IBM
 * Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import resources from '../../../../lib/shared/resources'
import {
  Accordion,
  AccordionItem,
  Checkbox,
  DropdownV2,
  Icon,
  RadioButton,
  RadioButtonGroup,
  TimePicker
} from 'carbon-components-react'
import { getSourcePath, removeVs } from '../../TemplateEditor/utils/utils'
import Tooltip from '../../TemplateEditor/components/Tooltip'
import msgs from '../../../../nls/platform.properties'
import _ from 'lodash'

resources(() => {
  require('./style.scss')
})

export class TimeWindow extends React.Component {
  static propTypes = {
    control: PropTypes.object,
    controlId: PropTypes.string,
    handleChange: PropTypes.func,
    locale: PropTypes.string
  };

  constructor(props) {
    super(props)
    this.state = {}
    if (_.isEmpty(this.props.control.active)) {
      this.props.control.active = {
        mode: '',
        days: [],
        timezone: '',
        showTimeSection: false,
        timeList: [{ id: 0, start: '', end: '', validTime: true }],
        timeListID: 1
      }
    }
    this.props.control.validation = this.validation.bind(this)
  }

  validation(exceptions) {
    const { control, locale } = this.props
    // Mode is active/blocked
    if (control.active.mode) {
      // Add exception if no days selected
      if (control.active.days.length === 0) {
        exceptions.push({
          row: 1,
          text: msgs.get('creation.missing.timeWindow.days', locale),
          type: 'error',
          controlId: 'timeWindow-config'
        })
      }
      // Add exception if no timezone selected
      if (control.active.timezone === '') {
        exceptions.push({
          row: 1,
          text: msgs.get('creation.missing.timeWindow.timezone', locale),
          type: 'error',
          controlId: 'timeWindow-config'
        })
      }
    }
  }

  render() {
    if (!this.props.control.active) {
      this.props.control.active = {
        mode: '',
        days: [],
        timezone: '',
        showTimeSection: false,
        timeList: [{ id: 0, start: '', end: '', validTime: true }],
        timeListID: 1
      }
    }
    const { controlId, locale, control } = this.props
    const { name, active, validation = {} } = control
    const modeSelected = active && active.mode ? true : false
    const daysSelectorID = 'days-selector'
    const timezoneDropdownID = 'timezone-dropdown'
    const { mode, days = [], timezone } = this.props.control.active

    return (
      <React.Fragment>
        <div className="creation-view-controls-labels">
          <div className="creation-view-controls-textarea-title">
            {name}
            {validation.required ? (
              <div className="creation-view-controls-required">*</div>
            ) : null}
            <Tooltip control={control} locale={locale} />
          </div>

          <div className="timeWindow-container">
            <RadioButtonGroup
              className="timeWindow-mode-container"
              name={`timeWindow-mode-container-${controlId}`}
              defaultSelected={mode ? `"${mode}"` : ''}
              id={controlId}
            >
              <RadioButton
                className="mode-btn"
                id={`default-mode-${controlId}`}
                labelText={msgs.get(
                  'creation.app.settings.timeWindow.defaultMode',
                  locale
                )}
                value=""
                onClick={this.handleChange.bind(this)}
              />
              <RadioButton
                className="mode-btn"
                id={`active-mode-${controlId}`}
                labelText={msgs.get(
                  'creation.app.settings.timeWindow.activeMode',
                  locale
                )}
                value="&quot;active&quot;"
                onClick={this.handleChange.bind(this)}
              />
              <RadioButton
                className="mode-btn"
                id={`blocked-mode-${controlId}`}
                labelText={msgs.get(
                  'creation.app.settings.timeWindow.blockedMode',
                  locale
                )}
                value="&quot;blocked&quot;"
                onClick={this.handleChange.bind(this)}
              />
            </RadioButtonGroup>

            <Accordion>
              <AccordionItem
                open={modeSelected}
                title={msgs.get(
                  'creation.app.settings.timeWindow.config',
                  locale
                )}
              >
                <div
                  className="timeWindow-config-container"
                  id="timeWindow-config"
                >
                  <div className="config-days-section">
                    <div className="config-title">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.days.title',
                        locale
                      )}{' '}
                      *
                    </div>
                    <div className="config-descr">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.days.descr',
                        locale
                      )}
                    </div>
                    <div className="config-days-selector">
                      <div className="first-col">
                        <Checkbox
                          checked={days.includes('"Monday"')}
                          labelText="Monday"
                          name={daysSelectorID}
                          id={`mon-${controlId}`}
                          value="&quot;Monday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          checked={days.includes('"Tuesday"')}
                          labelText="Tuesday"
                          name={daysSelectorID}
                          id={`tue-${controlId}`}
                          value="&quot;Tuesday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          checked={days.includes('"Wednesday"')}
                          labelText="Wednesday"
                          name={daysSelectorID}
                          id={`wed-${controlId}`}
                          value="&quot;Wednesday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          checked={days.includes('"Thursday"')}
                          labelText="Thursday"
                          name={daysSelectorID}
                          id={`thu-${controlId}`}
                          value="&quot;Thursday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          checked={days.includes('"Friday"')}
                          labelText="Friday"
                          name={daysSelectorID}
                          id={`fri-${controlId}`}
                          value="&quot;Friday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                      </div>
                      <div className="second-col">
                        <Checkbox
                          checked={days.includes('"Saturday"')}
                          labelText="Saturday"
                          name={daysSelectorID}
                          id={`sat-${controlId}`}
                          value="&quot;Saturday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          checked={days.includes('"Sunday"')}
                          labelText="Sunday"
                          name={daysSelectorID}
                          id={`sun-${controlId}`}
                          value="&quot;Sunday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                      </div>
                    </div>
                  </div>

                  <div className="config-timezone-section">
                    <div className="config-title">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.timezone.title',
                        locale
                      )}{' '}
                      *
                    </div>
                    <DropdownV2
                      className="config-timezone-dropdown"
                      label={timezone || 'Choose a location'}
                      items={[
                        {
                          label: 'America/Edmonton',
                          name: timezoneDropdownID,
                          value: '"America/Edmonton"'
                        },
                        {
                          label: 'America/Halifax',
                          name: timezoneDropdownID,
                          value: '"America/Halifax"'
                        },
                        {
                          label: 'America/Toronto',
                          name: timezoneDropdownID,
                          value: '"America/Toronto"'
                        },
                        {
                          label: 'America/Vancouver',
                          name: timezoneDropdownID,
                          value: '"America/Vancouver"'
                        },
                        {
                          label: 'America/Winnipeg',
                          name: timezoneDropdownID,
                          value: '"America/Winnipeg"'
                        }
                      ]}
                      disabled={!modeSelected}
                      onChange={this.handleChange.bind(this)}
                    />
                  </div>

                  <div className="config-time-section">
                    {this.renderTimes(control, modeSelected)}
                    <div
                      className={`add-time-btn ${
                        !modeSelected ? 'btn-disabled' : ''
                      }`}
                      tabIndex="0"
                      role={'button'}
                      onClick={() => this.addTimeToList(control, modeSelected)}
                      onKeyPress={this.addTimeKeyPress.bind(this)}
                    >
                      <Icon
                        name="icon--add--glyph"
                        fill="#3d70b2"
                        description=""
                        className="add-time-btn-icon"
                      />
                      <div className="add-time-btn-text">
                        {msgs.get(
                          'creation.app.settings.timeWindow.config.time.add',
                          locale
                        )}
                      </div>
                    </div>
                  </div>
                </div>
              </AccordionItem>
            </Accordion>
          </div>
        </div>
      </React.Fragment>
    )
  }

  renderTimes = (control, modeSelected) => {
    return (
      control.active &&
      control.active.timeList.map(item => {
        const { id, start, end, validTime } = item
        // Don't show deleted time invertals
        if (validTime) {
          return (
            <React.Fragment key={id}>
              <div className="config-time-container">
                <div className="config-start-time">
                  <TimePicker
                    id={`start-time-${id}`}
                    name="start-time"
                    labelText={id === 0 ? 'Start Time' : ''}
                    type="time"
                    value={this.state.startTime || to24(start) || ''}
                    disabled={!modeSelected}
                    onChange={this.handleChange.bind(this)}
                  />
                </div>
                <div className="config-end-time">
                  <TimePicker
                    id={`end-time-${id}`}
                    name="end-time"
                    labelText={id === 0 ? 'End Time' : ''}
                    type="time"
                    value={this.state.endTime || to24(end) || ''}
                    disabled={!modeSelected}
                    onChange={this.handleChange.bind(this)}
                  />
                </div>
                {id !== 0 ? ( // Option to remove added times
                  <div
                    id={id}
                    className="remove-time-btn"
                    tabIndex="0"
                    role={'button'}
                    onClick={() => this.removeTimeFromList(control, item)}
                    onKeyPress={this.removeTimeKeyPress.bind(this)}
                  >
                    <Icon
                      name="icon--close--glyph"
                      fill="#3d70b2"
                      className="remove-time-btn-icon"
                    />
                  </div>
                ) : (
                  ''
                )}
              </div>
            </React.Fragment>
          )
        }
        return ''
      })
    )
  };

  // Convert 24-hour format to 12-hour format
  convertTimeFormat = time => {
    if (time !== '') {
      const hour24 = +time.substring(0, 2)
      let hour12 = hour24 % 12 || 12
      hour12 = hour12 < 10 ? '0' + hour12 : hour12
      const period = hour24 < 12 ? 'AM' : 'PM'
      return hour12 + time.substring(2) + period
    } else {
      return ''
    }
  };

  addTimeToList = (control, modeSelected) => {
    if (modeSelected) {
      // Create new "time" item
      control.active.timeList.push({
        id: control.active.timeListID,
        start: '',
        end: '',
        validTime: true
      })
      control.active.timeListID++

      // Update UI
      this.forceUpdate()
    }
  };

  addTimeKeyPress = e => {
    if (e.type === 'click' || e.key === 'Enter') {
      this.addTimeToList(this.props.control)
    }
  };

  removeTimeFromList = (control, item) => {
    // Removed times are no longer valid
    control.active.timeList[item.id].validTime = false

    // Update UI and yaml editor
    this.forceUpdate()
    this.handleChange({})
  };

  removeTimeKeyPress = e => {
    if (e.type === 'click' || e.key === 'Enter') {
      this.removeTimeFromList(this.props.control, { id: e.target.id })
    }
  };

  handleChange(event) {
    const { control, handleChange } = this.props
    let targetName = ''
    try {
      targetName = event.target.name
    } catch (e) {
      targetName = ''
    }

    if (targetName) {
      if (targetName.startsWith('timeWindow-mode-container')) {
        control.active.mode = event.target.value
      } else {
        switch (targetName) {
        case 'days-selector':
          if (event.target.checked === true) {
            control.active.days.push(event.target.value)
          } else {
            const index = control.active.days.indexOf(event.target.value)
            control.active.days.splice(index, 1)
          }
          break
        case 'start-time':
          {
            const startTimeID = parseInt(event.target.id.split('-')[2], 10)
            const convertedTime = this.convertTimeFormat(event.target.value)
            // As long as first start-time is entered, all times will show
            if (startTimeID === 0) {
              control.active.showTimeSection = convertedTime ? true : false
            }
            control.active.timeList[startTimeID].start = convertedTime
          }
          break
        case 'end-time':
          {
            const endTimeID = parseInt(event.target.id.split('-')[2], 10)
            control.active.timeList[endTimeID].end = this.convertTimeFormat(
              event.target.value
            )
          }
          break
        }
      }
    } else if (
      event.selectedItem &&
      event.selectedItem.name === 'timezone-dropdown'
    ) {
      control.active.timezone = event.selectedItem.value
    }

    handleChange(control)
  }
}

export default TimeWindow

export const reverse = (control, templateObject) => {
  let showTimeSection = false
  const timezone = _.get(
    templateObject,
    getSourcePath('Subscription[0].spec.timewindow.location')
  )
  const mode = _.get(
    templateObject,
    getSourcePath('Subscription[0].spec.timewindow.windowtype')
  )
  let weekdays = _.get(
    templateObject,
    getSourcePath('Subscription[0].spec.timewindow.weekdays')
  )
  weekdays = (removeVs(weekdays && weekdays.$v) || []).map(day => {
    return `"${day}"`
  })
  let timeList = _.get(
    templateObject,
    getSourcePath('Subscription[0].spec.timewindow.hours')
  )
  if (timeList) {
    timeList = removeVs(timeList)
  }
  if (timeList) {
    timeList = timeList.map(({ start, end }, id) => {
      return {
        id,
        start,
        end,
        validTime: true
      }
    })
    showTimeSection = true
  } else {
    timeList = [{ id: 0, start: '', end: '', validTime: true }]
  }
  control.active = {
    mode: mode && mode.$v,
    days: weekdays,
    timezone: timezone && timezone.$v,
    showTimeSection,
    timeList,
    timeListID: 1
  }
}

// Convert 12-hour format to 24-hour format
const to24 = time => {
  const match = /((1[0-2]|0?[1-9]):([0-5][0-9])([AP][M]))/.exec(time)
  if (match) {
    const [, , hour12, minute, period] = match
    let hour = parseInt(hour12, 10)
    if (hour < 12 && period === 'PM') {
      hour += 12
    }
    if (hour < 10) {
      hour = `0${hour}`
    }
    return `${hour}:${minute}`
  }
  return time
}
