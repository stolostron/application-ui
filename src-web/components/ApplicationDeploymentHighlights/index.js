/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

import R from 'ramda'
import React from 'react'

import { connect } from 'react-redux'
import resources from '../../../lib/shared/resources'
import ApplicationDeploymentHighlightsTerminology from './ApplicationDeploymentHighlightsTerminology'

/* eslint-disable react/prop-types */

resources(() => {
  require('./style.scss')
})

const mapStateToProps = state => {
  const { QueryApplicationList, secondaryHeader } = state
  return {
    QueryApplicationList,
    secondaryHeader
  }
}

class ApplicationDeploymentHighlights extends React.Component {
  render() {
    const { QueryApplicationList, secondaryHeader } = this.props

    const applications = R.pathOr([], ['items'])(QueryApplicationList)
    let open = applications.length === 0
    if (applications.length > 0) {
      const isSingleApplicationView =
        R.pathOr([], ['breadcrumbItems'])(secondaryHeader).length === 2
      let selectedAppName = ''
      let selectedAppNS = ''

      if (isSingleApplicationView) {
        const urlArray = R.split('/', secondaryHeader.breadcrumbItems[1].url)
        selectedAppName = urlArray[urlArray.length - 1]
        selectedAppNS = urlArray[urlArray.length - 2]

        const apps = R.find(
          R.propEq('name', selectedAppName) &&
            R.propEq('namespace', selectedAppNS)
        )(applications) //filter by name, ns
        open = !(
          apps &&
          apps.hubSubscriptions &&
          apps.hubSubscriptions.length > 0
        )
      }
    }

    return (
      <div id="DeploymentHighlights">
        <ApplicationDeploymentHighlightsTerminology open={open} />
      </div>
    )
  }
}

export default connect(mapStateToProps)(ApplicationDeploymentHighlights)
