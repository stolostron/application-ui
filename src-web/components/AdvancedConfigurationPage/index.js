/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.

import React from 'react'
import msgs from '../../../nls/platform.properties'
import { connect } from 'react-redux'
import resources from '../../../lib/shared/resources'
import {
  updateSecondaryHeader,
  clearSuccessFinished
} from '../../actions/common'

import ApplicationDeploymentHighlights from '../ApplicationDeploymentHighlights'
import { withRouter } from 'react-router-dom'
import AdvancedConfigurationLists from '../AdvancedConfigurationLists'

/* eslint-disable react/prop-types */

resources(() => {
  require('./style.scss')
})

const mapDispatchToProps = dispatch => {
  return {
    clearSuccessFinished: () => clearSuccessFinished(dispatch),
    updateSecondaryHeaderFn: (title, tabs, mainButton) =>
      dispatch(
        updateSecondaryHeader(title, tabs, null, null, null, null, mainButton)
      )
  }
}

class AdvancedConfigurationPage extends React.Component {
  componentDidMount() {
    const {
      secondaryHeaderProps,
      updateSecondaryHeaderFn,
      locale
    } = this.props

    updateSecondaryHeaderFn(
      msgs.get(secondaryHeaderProps.title, locale),
      secondaryHeaderProps.tabs,
      secondaryHeaderProps.mainButton
    )
  }

  componentWillUnmount() {
    this.mutateFinished()
  }

  mutateFinished() {
    this.props.clearSuccessFinished()
  }

  render() {
    const { secondaryHeaderProps } = this.props

    return (
      <React.Fragment>
        <ApplicationDeploymentHighlights />
        <AdvancedConfigurationLists
          secondaryHeaderProps={secondaryHeaderProps}
        />
      </React.Fragment>
    )
  }
}

export default withRouter(
  connect(null, mapDispatchToProps)(AdvancedConfigurationPage)
)
