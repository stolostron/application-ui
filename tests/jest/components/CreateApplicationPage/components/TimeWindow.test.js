/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

"use strict";

import React from "react";
import { TimeWindow } from "../../../../../src-web/components/ApplicationCreationPage/components/TimeWindow";
import renderer from "react-test-renderer";
import _ from "lodash";
import { mount } from "enzyme";

const control = {
  active: {}
};

describe("TimeWindow component", () => {
  it("renders as expected", () => {
    const component = renderer.create(
      <TimeWindow
        locale="en-US"
        type="custom"
        available={[]}
        control={control}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("on control change function", () => {
  it("renders as expected", () => {
    const wrapper = mount(
      <TimeWindow
        locale="en-US"
        type="custom"
        available={[]}
        control={control}
      />
    );
    const evt = {
      target: {
        value: "value-testing"
      },
      selectedItems: ["selectedItems-testing-1", "selectedItems-testing-2"]
    };

    wrapper
      .find("#default-mode")
      .at(0)
      .simulate("click", evt);
    wrapper
      .find("#active-mode")
      .at(0)
      .simulate("click", evt);
    wrapper
      .find("#block-mode")
      .at(0)
      .simulate("click", evt);
    wrapper
      .find(".bx--accordion__item")
      .at(0)
      .simulate("keypress", evt);

    wrapper
      .find(".bx--accordion__heading")
      .at(0)
      .simulate("click", evt);

    wrapper
      .find("#tues")
      .at(0)
      .simulate("click", evt);
  });
});
