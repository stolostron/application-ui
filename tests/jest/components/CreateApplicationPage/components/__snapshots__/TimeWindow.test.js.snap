// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeWindow component renders as expected 1`] = `
<div
  className="creation-view-controls-labels"
>
  <div
    className="creation-view-controls-textarea-title"
  />
  <div
    className="timeWindow-container"
  >
    <div
      className="bx--form-item"
    >
      <div
        className="timeWindow-mode-container"
      >
        <div
          className="radioButtonWrapper mode-btn"
        >
          <input
            checked={true}
            className="bx--radio-button"
            id="default-mode-undefined"
            name="timeWindow-mode-container-undefined"
            onChange={[Function]}
            onClick={[Function]}
            type="radio"
            value=""
          />
          <label
            className="bx--radio-button__label"
            htmlFor="default-mode-undefined"
          >
            <span
              className="bx--radio-button__appearance"
            />
            Always active
          </label>
        </div>
        <div
          className="radioButtonWrapper mode-btn"
        >
          <input
            checked={false}
            className="bx--radio-button"
            id="active-mode-undefined"
            name="timeWindow-mode-container-undefined"
            onChange={[Function]}
            onClick={[Function]}
            type="radio"
            value="\\"active\\""
          />
          <label
            className="bx--radio-button__label"
            htmlFor="active-mode-undefined"
          >
            <span
              className="bx--radio-button__appearance"
            />
            Active within specified interval
          </label>
        </div>
        <div
          className="radioButtonWrapper mode-btn"
        >
          <input
            checked={false}
            className="bx--radio-button"
            id="blocked-mode-undefined"
            name="timeWindow-mode-container-undefined"
            onChange={[Function]}
            onClick={[Function]}
            type="radio"
            value="\\"blocked\\""
          />
          <label
            className="bx--radio-button__label"
            htmlFor="blocked-mode-undefined"
          >
            <span
              className="bx--radio-button__appearance"
            />
            Blocked within specified interval
          </label>
        </div>
      </div>
    </div>
    <ul
      className="bx--accordion"
    >
      <li
        className="bx--accordion__item"
        onClick={[Function]}
        onKeyPress={[Function]}
        open={false}
        role="presentation"
      >
        <button
          aria-expanded={false}
          className="bx--accordion__heading"
          onClick={[Function]}
          type="button"
        >
          <svg
            alt="Expand/Collapse"
            aria-label="Expand/Collapse"
            className="bx--accordion__arrow"
            fillRule="evenodd"
            height="12"
            role={null}
            viewBox="0 0 7 12"
            width="7"
          >
            <title>
              Expand/Collapse
            </title>
            <path
              d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
            />
          </svg>
          <div
            className="bx--accordion__title"
          >
            Time window configuration
          </div>
        </button>
        <div
          className="bx--accordion__content"
        >
          <div
            className="timeWindow-config-container"
            id="timeWindow-config"
          >
            <div
              className="config-days-section"
            >
              <div
                className="config-title"
              >
                Days of the week
                 
                *
              </div>
              <div
                className="config-descr"
              >
                Select at least one day to create a time window.
              </div>
              <div
                className="config-days-selector"
              >
                <div
                  className="first-col"
                >
                  <div
                    className="bx--form-item bx--checkbox-wrapper"
                  >
                    <input
                      checked={false}
                      className="bx--checkbox"
                      disabled={true}
                      id="mon-undefined"
                      name="days-selector"
                      onChange={[Function]}
                      onClick={[Function]}
                      type="checkbox"
                      value="\\"Monday\\""
                    />
                    <label
                      className="bx--checkbox-label"
                      htmlFor="mon-undefined"
                      title={null}
                    >
                      <span
                        className=""
                      >
                        Monday
                      </span>
                    </label>
                  </div>
                  <div
                    className="bx--form-item bx--checkbox-wrapper"
                  >
                    <input
                      checked={false}
                      className="bx--checkbox"
                      disabled={true}
                      id="tue-undefined"
                      name="days-selector"
                      onChange={[Function]}
                      onClick={[Function]}
                      type="checkbox"
                      value="\\"Tuesday\\""
                    />
                    <label
                      className="bx--checkbox-label"
                      htmlFor="tue-undefined"
                      title={null}
                    >
                      <span
                        className=""
                      >
                        Tuesday
                      </span>
                    </label>
                  </div>
                  <div
                    className="bx--form-item bx--checkbox-wrapper"
                  >
                    <input
                      checked={false}
                      className="bx--checkbox"
                      disabled={true}
                      id="wed-undefined"
                      name="days-selector"
                      onChange={[Function]}
                      onClick={[Function]}
                      type="checkbox"
                      value="\\"Wednesday\\""
                    />
                    <label
                      className="bx--checkbox-label"
                      htmlFor="wed-undefined"
                      title={null}
                    >
                      <span
                        className=""
                      >
                        Wednesday
                      </span>
                    </label>
                  </div>
                  <div
                    className="bx--form-item bx--checkbox-wrapper"
                  >
                    <input
                      checked={false}
                      className="bx--checkbox"
                      disabled={true}
                      id="thu-undefined"
                      name="days-selector"
                      onChange={[Function]}
                      onClick={[Function]}
                      type="checkbox"
                      value="\\"Thursday\\""
                    />
                    <label
                      className="bx--checkbox-label"
                      htmlFor="thu-undefined"
                      title={null}
                    >
                      <span
                        className=""
                      >
                        Thursday
                      </span>
                    </label>
                  </div>
                  <div
                    className="bx--form-item bx--checkbox-wrapper"
                  >
                    <input
                      checked={false}
                      className="bx--checkbox"
                      disabled={true}
                      id="fri-undefined"
                      name="days-selector"
                      onChange={[Function]}
                      onClick={[Function]}
                      type="checkbox"
                      value="\\"Friday\\""
                    />
                    <label
                      className="bx--checkbox-label"
                      htmlFor="fri-undefined"
                      title={null}
                    >
                      <span
                        className=""
                      >
                        Friday
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  className="second-col"
                >
                  <div
                    className="bx--form-item bx--checkbox-wrapper"
                  >
                    <input
                      checked={false}
                      className="bx--checkbox"
                      disabled={true}
                      id="sat-undefined"
                      name="days-selector"
                      onChange={[Function]}
                      onClick={[Function]}
                      type="checkbox"
                      value="\\"Saturday\\""
                    />
                    <label
                      className="bx--checkbox-label"
                      htmlFor="sat-undefined"
                      title={null}
                    >
                      <span
                        className=""
                      >
                        Saturday
                      </span>
                    </label>
                  </div>
                  <div
                    className="bx--form-item bx--checkbox-wrapper"
                  >
                    <input
                      checked={false}
                      className="bx--checkbox"
                      disabled={true}
                      id="sun-undefined"
                      name="days-selector"
                      onChange={[Function]}
                      onClick={[Function]}
                      type="checkbox"
                      value="\\"Sunday\\""
                    />
                    <label
                      className="bx--checkbox-label"
                      htmlFor="sun-undefined"
                      title={null}
                    >
                      <span
                        className=""
                      >
                        Sunday
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="config-timezone-section"
            >
              <div
                className="config-title"
              >
                Time zone
                 
                *
              </div>
              <div
                aria-label="Choose an item"
                className="bx--combo-box config-timezone-combo-box bx--list-box bx--list-box--disabled"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="listbox"
                tabIndex="0"
              >
                <div
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-label="open menu"
                  className="bx--list-box__field"
                  data-toggle={true}
                  disabled={true}
                  role="button"
                  tabIndex="0"
                  type="button"
                >
                  <input
                    aria-activedescendant={null}
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-label="timezoneComboBox"
                    autoComplete="off"
                    className="bx--text-input"
                    disabled={true}
                    id="downshift-0-input"
                    label="Time zone combo box menu options"
                    placeholder="Choose a location"
                    role="combobox"
                    value=""
                  />
                  
                  <div
                    className="bx--list-box__menu-icon"
                  >
                    <svg
                      alt="Open menu"
                      aria-label="Open menu"
                      fillRule="evenodd"
                      height="5"
                      role="img"
                      viewBox="0 0 10 5"
                      width="10"
                    >
                      <title>
                        Open menu
                      </title>
                      <path
                        d="M0 0l5 4.998L10 0z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="config-time-section"
            >
              <div
                className="config-time-container"
              >
                <div
                  className="config-start-time"
                >
                  <div
                    className="bx--form-item"
                  >
                    <div
                      className="bx--time-picker"
                    >
                      <div
                        className="bx--time-picker__input"
                      >
                        <label
                          className="bx--label"
                          htmlFor="start-time-0"
                        >
                          Start Time
                        </label>
                        <input
                          className="bx--time-picker__input-field"
                          disabled={true}
                          id="start-time-0"
                          maxLength={5}
                          name="start-time"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          pattern="(1[012]|[1-9]):[0-5][0-9](\\\\s)?(?i)"
                          placeholder="hh:mm"
                          type="time"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="config-end-time"
                >
                  <div
                    className="bx--form-item"
                  >
                    <div
                      className="bx--time-picker"
                    >
                      <div
                        className="bx--time-picker__input"
                      >
                        <label
                          className="bx--label"
                          htmlFor="end-time-0"
                        >
                          End Time
                        </label>
                        <input
                          className="bx--time-picker__input-field"
                          disabled={true}
                          id="end-time-0"
                          maxLength={5}
                          name="end-time"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          pattern="(1[012]|[1-9]):[0-5][0-9](\\\\s)?(?i)"
                          placeholder="hh:mm"
                          type="time"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                
              </div>
              <div
                className="add-time-btn btn-disabled"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                <svg
                  alt=""
                  aria-label=""
                  className="add-time-btn-icon"
                  fill="#3d70b2"
                  fillRule="evenodd"
                  height="16"
                  name="icon--add--glyph"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <title>
                    
                  </title>
                  <path
                    d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z"
                  />
                </svg>
                <div
                  className="add-time-btn-text"
                >
                  Add another time range
                </div>
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
